Ci regression from  #69746

**Describe the bug**
https://github.com/zephyrproject-rtos/zephyr/pull/69746 broke the nanopb build in CI for most platforms:

```
$ west build -p -b nrf52dk/nrf52832 samples/modules/nanopb
...
CMake Error at /usr/local/google/home/fabiobaltieri/zephyrproject/modules/lib/nanopb/extra/FindNanopb.cmake:407 (target_compile_features):
  target_compile_features The compiler feature "c_std_11" is not known to C
  compiler

  "GNU"
```

https://github.com/zephyrproject-rtos/zephyr/actions/runs/8364553686/job/22900251867

**To Reproduce**
`west build -p -b nrf52dk/nrf52832 samples/modules/nanopb` on main

**Expected behavior**
Not break.

**Impact**
The "push" CI run is very red, does not seem to be impacting pull request targets.

--- a/submanifests/optional.yaml
+++ b/submanifests/optional.yaml
@@ -22,7 +22,7 @@ manifest:
       groups:
         - optional
     - name: nanopb
-      revision: 671672b4d7994a9b07a307ae654885c7202ae886
+      revision: 65cbefb4695bc7af1cb733ced99618afb3586b20
       path: modules/lib/nanopb
       remote: upstream
       groups:

